core/src/main/scala-2.13+/cats/instances/lazyList.scala

```diff
@@ -122,6 +127,27 @@ trait LazyListInstances extends cats.kernel.instances.LazyListInstances {
 
       override def collectFirstSome[A, B](fa: LazyList[A])(f: A => Option[B]): Option[B] =
         fa.collectFirst(Function.unlift(f))
+
+      def functor: Functor[LazyList] = this
+
+      def align[A, B](fa: LazyList[A], fb: LazyList[B]): LazyList[Ior[A, B]] =
+        alignWith(fa, fb)(identity)
+
+      override def alignWith[A, B, C](fa: LazyList[A], fb: LazyList[B])(f: Ior[A, B] => C): LazyList[C] = {
+        val iterA = fa.iterator
+        val iterB = fb.iterator
+
+        var result: LazyList[C] = LazyList.empty
+
+        while (iterA.hasNext || iterB.hasNext) {
```

Comment:
This will trigger evaluation of all items right? I imagine we should be able to write a lazy implementation right?

Reply:
Oh does it? I haven't actually looked at how `LazyList` works at all. Help welcome :)

Reply:
me neither. but I think you can create a new iterator something like 
```
val alighIterator = new Iterator[C] {
    def hasNext: Boolean = iterA.hasNext || iterB.hasNext
    def next(): C   = 
            val ior =
               if (iterA.hasNext && iterB.hasNext) Ior.both(iterA.next(), iterB.next())
               else if (iterA.hasNext) Ior.left(iterA.next())
               else Ior.right(iterB.next())
           f(ior)
}

LazyList.from(alignIterator)
```

Reply:
Thanks this seems to work just fine :)